holoviews/plotting/bokeh/util.py

```diff
@@ -547,3 +547,46 @@ def get_tab_title(key, frame, overlay):
         title = ' | '.join([d.pprint_value_string(k) for d, k in
                             zip(overlay.kdims, key)])
     return title
+
+
+def expand_batched_style(style, opts, data, mapping, nvals=None, multiple=False):
+    """
+    Computes styles applied to a batched plot by iterating over the
+    supplied list of style options and expanding any options found in
+    the supplied style dictionary to the ColumnDataSource data and
+    mapping. Supplying nvals adds the style entry multiple times, added
+    as an array by default. When multiple is active multiple scalar values
+    will be added.
+    """
```

Comment:
Yeah, it's fairly complex I'll try to come up with some examples, since it supports three modes of expanding styles corresponding to Path, Points, and Curve types.